kcp/controllers/kyma_controller_test.go

```diff
@@ -0,0 +1,144 @@
+package controllers_test
+
+import (
+	"time"
+
+	"github.com/kyma-project/kyma-operator/operator/api/v1alpha1"
+	componentv1alpha1 "github.com/kyma-project/kyma-watcher/kcp/api/v1alpha1"
+	"github.com/kyma-project/kyma-watcher/kcp/controllers"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	namespace        = "default"
+	contractVersion  = "v1"
+	componentName    = "example-module"
+	componentChannel = "stable"
+	serviceName      = "test-service"
+	servicePort      = 8083
+	watcherName      = componentName + "-" + componentChannel
+	configMapName    = componentName + "-" + componentChannel
+
+	interval = time.Millisecond * 250
+)
+
+var _ = Describe("Correct WatcherCR Setup", func() {
+	testKyma := NewTestKyma("test-kyma")
+	testConfigMap := NewTestConfigMap(configMapName)
+	testWatcherCR := NewTestWatcherCR(watcherName, map[string]string{controllers.DefaultOperatorWatcherCRLabel: "true"})
+
+	SetupTestEnvironment(testKyma, testConfigMap, testWatcherCR)
+
+	It("should insert testKyma in the ConfigMap of the example-module WatcherCR", func() {
+		By("checking the data of the ConfigMap")
+		Eventually(GetConfigMapData(testConfigMap), 5*time.Second, interval).
+			Should(Equal(map[string]string{"test-kyma": "default"}))
+	})
+})
+
+var _ = Describe("WatcherCR applies to all Kymas - Configmap stores no data", func() {
+	testKyma := NewTestKyma("test-kyma")
+	testConfigMap := NewTestConfigMap(configMapName)
+	testWatcherCR := NewTestWatcherCR(watcherName, map[string]string{})
+
+	SetupTestEnvironment(testKyma, testConfigMap, testWatcherCR)
+
+	It("should not throw an error in reconcile loop an ConfigMap should not be initialised nor updated", func() {
+		By("checking the data of the ConfigMap")
+		Eventually(GetConfigMapData(testConfigMap), 5*time.Second, interval).Should(BeEmpty())
+	})
+})
```

Comment:
If you wish: I think these two tests can go to once spec, for simplicity

Reply:
I can rewrite them, so they will fit in one spec